source/extensions/matching/input_matchers/ip/matcher.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <vector>
+
+#include "envoy/matcher/matcher.h"
+#include "envoy/network/address.h"
+
+#include "common/network/lc_trie.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Matching {
+namespace InputMatchers {
+namespace IP {
+
+class Matcher : public Envoy::Matcher::InputMatcher {
+public:
+  Matcher(std::vector<Network::Address::CidrRange>&& ranges);
+  bool match(absl::optional<absl::string_view> input) override;
+
+private:
+  Network::LcTrie::LcTrie<bool> trie_;
```

Comment:
const?

Reply:
Indeed! Added in https://github.com/envoyproxy/envoy/pull/16592/commits/a0360a73c6b3df1e3a75191f813939fe8ce49608